add solution

VadymVynokur opened the pull request.

VadymVynokur committed:
add solution

petrykivd commented on app/main.py:
```diff
@@ -1,3 +1,62 @@
+
+    def __add__(self, other: int) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
```
other can be not only int :)

petrykivd replied:
Check type hints in all program

petrykivd commented on app/main.py:
```diff
@@ -1,3 +1,62 @@
+    def __add__(self, other: int) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        else:
```
Also you can remove else here, first return will finish func execution, if not func continue witout else construction

petrykivd commented on app/main.py:
```diff
@@ -1,3 +1,62 @@
+            return self
+        else:
+            self.km += other
+            return self
```
Same here about type hinting

petrykivd replied:
try to rewrite without having return duplicated

petrykivd commented on app/main.py:
```diff
@@ -1,3 +1,62 @@
+        if isinstance(other, Distance):
+            return self.km >= other.km
+        else:
+            return self.km >= other
```
Let's simplify without if else construction, remember, you already implemented lt and gt methods

petrykivd commented on app/main.py:
```diff
@@ -1,3 +1,62 @@
+        if isinstance(self, Distance):
+            return Distance(round(self.km / other, 2))
+        else:
+            raise TypeError("Cannot divide distance123")
```
Fix with f string

VadymVynokur committed:
fix type hints in all program
removed else in __add__
fix return duplicating in __iadd__
simplify __le__ __ge__, by using __lt__ and __gt__
removed useless if/else/raise , idk why it was there, it is not part of task

petrykivd commented on app/main.py:
```diff
@@ -1,3 +1,55 @@
+        return self.__lt__(other) or self.__eq__(other)
+
+    def __ge__(self, other: Distance | int) -> bool:
+        return self.__gt__(other) or self.__eq__(other)
```
Replace method calls with < , >  😉

petrykivd commented on app/main.py:
```diff
@@ -1,3 +1,55 @@
+        else:
+            return self.km > other
+
+    def __eq__(self, other: Distance | int) -> bool:
```
What about floats in type hints?:)

VadymVynokur replied:
float were not mentioned in the task condition, or rather their mention was only for __truediv__

petrykivd replied:
So we can first divide the distance and then add it to the other one, in which case other can be float? :)

Also please before pushing your solution, read [https://github.com/mate-academy/py-distance-class/blob/master/checklist.md](checklist) 😉

VadymVynokur committed:
make __le__ and __ge__ more readable

petrykivd reviewed (approved):
I approve this, but please correct the changes I requested 😃
